main.go

```diff
@@ -0,0 +1,67 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"unicode"
+)
+
+func IsSep(r rune) bool { return unicode.IsSpace(r) || unicode.IsPunct(r) || unicode.IsSymbol(r) }
+func words(s string) []string {
+	var w []string
+	var tmp []rune
+	wasSep := true
+	for _, r := range s {
+		if IsSep(r) {
+			if !wasSep {
+				w = append(w, string(tmp))
+
+				tmp = nil
+			}
+			wasSep = true
+		} else {
+			wasSep = false
+			tmp = append(tmp, r)
+		}
+	}
+	if len(tmp) > 0 {
+		w = append(w, string(tmp))
+	}
+	return w
+}
+
+func find(word string, words []string) int {
+	for s, w := range words {
```

Comment:
This is an index. For variables denoting indices, it's better to use names `i`, `j`, `k`.